app/controllers/users_controller.rb

```diff
@@ -65,4 +74,25 @@ def should_display_user?(user)
   def get_user
     @user = User.friendly.find(params[:id])
   end
+
+  def set_filter_params
+    filter_params = params.slice(:project_filter, :timezone_filter, :online)
+    set_timezone_degree_range(filter_params)
+    filter_params
+  end
+
+  def set_timezone_degree_range(filter_params)
+    unless filter_params[:timezone_filter].blank?
+      if lng = @current_user.try(:longitude)
+        case filter_params[:timezone_filter]
+        when 'Close To My Timezone Area'
+          filter_params[:timezone_filter] = [lng - 7.5, lng + 7.5]
+        when 'Wider Timezone Area'
+          filter_params[:timezone_filter] = [lng - 22.5, lng + 22.5]
+        end
+      else
+        redirect_to :back, alert: "Can't determine your location!"
```

Comment:
with regard to my previous comment, my guess is that this is the issue. `render` is still being called regardless of whether the `redirect` takes place. Could the `redirect` logic be moved inside the controller method or have the private methods renamed to indicate that a redirect might take place? It is hard to see where the response goes if the private methods are hiding a `redirect`.